dev/deps/spark-deps-hadoop-2.7

```diff
@@ -95,12 +95,15 @@ jackson-module-jaxb-annotations-2.9.9.jar
 jackson-module-paranamer-2.9.9.jar
 jackson-module-scala_2.12-2.9.9.jar
 jackson-xc-1.9.13.jar
+jakarta.activation-1.2.1.jar
+jakarta.activation-api-1.2.1.jar
```

Comment:
@srowen . We found that our library have difference dependency on JDK8 and JDK11.
I also noticed that this `jakarta.activation-1.2.1` occurs only on JDK11 environment.

I'm not sure how this library does this, but in this case, is it possible for Apache Spark to provide a single artifact for JDK8/JDK11? It seems that `Maven` artifacts might be possible, but the pre-built Apache Spark binary should be different due to this from now.

cc @dbtsai , @gatorsmile

Reply:
https://issues.apache.org/jira/browse/SPARK-28765 is filed. In case of `javax.annotation-api-1.2.jar`, the following is verified.
```
$ mvn dependency:tree -Dincludes=javax.annotation:javax.annotation-api -Phadoop-3.2 -Phive-thriftserver -Pmesos -Pkubernetes -Pyarn -Phive
...
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ spark-kubernetes_2.12 ---
[INFO] org.apache.spark:spark-kubernetes_2.12:jar:3.0.0-SNAPSHOT
[INFO] \- io.fabric8:kubernetes-client:jar:4.1.2:compile
[INFO]    \- io.fabric8:kubernetes-model:jar:4.1.2:compile
[INFO]       \- javax.annotation:javax.annotation-api:jar:1.2:compile
```

Reply:
Out of curiosity, is it an option to have two releases?

- Spark, Hadoop 2.7, JDK 8
- Spark, Hadoop 3.2, JDK 11

Since we currently skipped JDK 9 and 10 and JDK 8 is what we should drop later in the future anyway.

Reply:
It seems that Maven or Ivy bugs on JDK. After some investigation, the dependency seems not real. I'll make a PR very soon.

Reply:
I made a PR. Please take a look~
- https://github.com/apache/spark/pull/25481